test/Libraries/DynamoMSOfficeTests/ExcelTests.cs

```diff
@@ -935,6 +1182,351 @@ public void TestWriteFunctionObjectToExcel()
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            var watch = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("32dbe300-4780-4c47-b07c-63019d5285ac");
+            ViewModel.HomeSpace.Run();
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            var list2 = watch.CachedValue.GetElements();
+            Assert.AreEqual(5, list2.Count());
+            //   10, 20, 30, "test", "Dynamo"
+            Assert.IsTrue(list2[0].IsCollection);
+            var rowList1 = list2[0].GetElements();
+            Assert.AreEqual(1, rowList1[0].Data);
+            var rowList2 = list2[1].GetElements();
+            Assert.AreEqual(2, rowList2[0].Data);
+            var rowList3 = list2[2].GetElements();
+            Assert.AreEqual(3, rowList3[0].Data);
+            var rowList4 = list2[3].GetElements();
+            Assert.AreEqual(4, rowList4[0].Data);
+            var rowList5 = list2[4].GetElements();
+            Assert.AreEqual(5, rowList5[0].Data);
+        }
+        [Test,Category("Failure")]
+        public void WritenodeWithWarning()
+        {
+            // Write a node with warning to Excel.
+            // If there is any data in excel it must not overwrite here- currently this fails
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteNodewithError.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+            var watch = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("32dbe300-4780-4c47-b07c-63019d5285ac");
+            ViewModel.HomeSpace.Run();
+            var list = watch.CachedValue.Data;
+            Assert.IsNotNull(list);
+        }
+        [Test,Category("Failure")]
+        public void WriteToReadOnlyFile()
+        {
+
+            // asks for overwriting of file- need to find a way to ignore that.
+
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteToReadOnlyFile.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            var watch = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.WriteToFile");
+
+            var a = watch.Name;
+            var b = watch.InputNodes.Values;
```

Comment:
you need to make sure the `.xlsx` file is indeed readonly. I was able to run this successfully at my end.